sfa_dash/blueprints/util.py

```diff
@@ -25,35 +25,69 @@ def creation_link(cls, data_type, site_id=None):
         if site_id is not None:
             return url_for(f'forms.create_{data_type}', uuid=site_id)
         else:
-            return url_for('data_dashboard.sites', create=data_type)
+            return None
 
     @classmethod
-    def create_table_elements(cls, data_list, id_key, **kwargs):
+    def create_observation_table_elements(cls, data_list, **kwargs):
         """Creates a list of objects to be rendered as table by jinja template
         """
         sites_list = handle_response(sites.list_metadata())
-        site_dict = {site['site_id']: site for site in sites_list}
+        sites_dict = {site['site_id']: site for site in sites_list}
         table_rows = []
         for data in data_list:
             table_row = {}
-            site = site_dict.get(data['site_id'])
+            if data['site_id'] is not None:
+                site_id = data['site_id']
+                site = sites_dict.get(site_id)
             if site is not None:
-                site_name = site_dict[data['site_id']]['name']
+                site_name = site['name']
                 site_href = url_for('data_dashboard.site_view',
-                                    uuid=data['site_id'])
-                site_link = f'<a href={site_href}>{site_name}</a>'
+                                    uuid=site_id)
+                site_link = f'<a href="{site_href}">{site_name}</a>'
             else:
                 site_link = 'Site Unavailable'
             table_row['name'] = data['name']
             table_row['variable'] = data['variable']
             table_row['provider'] = data.get('provider', '')
             table_row['site'] = site_link
-            if id_key == 'forecast_id':
-                table_row['link'] = url_for('data_dashboard.forecast_view',
-                                            uuid=data[id_key])
+            table_row['link'] = url_for('data_dashboard.observation_view',
+                                        uuid=data['observation_id'])
+            table_rows.append(table_row)
+        return table_rows
+
+    @classmethod
+    def create_table_elements(cls, data_list, view_name, **kwargs):
```

Comment:
seems like these don't need to be separated. 

before the loop, if observations, don't get aggregates, and set the no location text ('Site unavailable').
Then modify the 
```python
if data.get('site_id') is not None:
   blah
elif data.get('aggregate_id') is not None:
   blah
else:
   location_id = None
```